### ISSUE TITLE
Shadow with blending: transparent / fade

### ISSUE THREAD

**Issue by michaelvogt** [google-ar/sceneform-android-sdk#535]
Using the sample code from #390, I see the following effect.

https://photos.app.goo.gl/VirWcNSyD8egTtw28
(sorry, uploading the screenshots didn't work)

As you can see, the shadow (and the surface dots, already reported, never fixed) is showing through the model. It has this custom material:
 
```
material {
    name: "Custom material",
    requires: [
        "position",
        "uv0"
    ],
    blending : "fade"
}

fragment {
    void material(inout MaterialInputs material) {
        prepareMaterial(material);

        material.baseColor.rgba = vec4(1.0, 1.0, 1.0, 1.0);
    }
}
```

Since alpha is set to 1.0, my expectation is, that the shadow (and of course the dots) should not be showing through. 

Also, the shape of the shadow seems kinda strange to me. I'd expect it to stay flat on this flat surface.

**Comment by romainguy** [comment]
The shadow is correct, it looks like this because your model intersects with the shadow plane. The problem here is not that the shadow is showing through the object but that the plane is drawn after your object.

**Comment by michaelvogt** [comment]
Here we go again. Not starting a discussion this time. I just voiced my observations.

> The problem here is not that the shadow is showing through the object but that the plane is drawn after your object

Are you saying there is something wrong with my code? I'm always kinda irritated with your answers.

I personally don't care what's wrong, I just want it fixed.

**Comment by romainguy** [comment]
Michael, I'm sorry my answers irritated you, it was definitely not my intention. I don't know whether the issue is in your code or not at this point (I'd be surprised if it is), I am just adding information to help with the debugging and fixing of this issue by the Sceneform team.

**Comment by michaelvogt** [comment]
> I'm sorry my answers irritated you, it was definitely not my intention.

Yes, I know, of course. Good to know that you are adding information to the Sceneform team with your answers.

**Comment by michaelvogt** [comment]
@romainguy 

Can you ping the sceneform team in some way? More than 2 weeks now without a reaction.

There is definitely something basic wrong with the rendering, that should be worth to fix.
